runtime/fiber.c

```diff
@@ -75,10 +78,25 @@ struct stack_info** caml_alloc_stack_cache (void)
 
 Caml_inline struct stack_info* alloc_for_stack (mlsize_t wosize)
 {
+#ifdef __OpenBSD__
+  struct stack_info* si;
+  size_t len = sizeof(struct stack_info) +
+               sizeof(value) * wosize +
+               8 /* for alignment */ +
+               sizeof(struct stack_handler);
+  si =  mmap(NULL, len, PROT_WRITE | PROT_READ,
+             MAP_ANONYMOUS | MAP_PRIVATE | MAP_STACK, -1, 0);
+  if (si == MAP_FAILED)
+    return NULL;
+
+  si->size = len;
+  return si;
+#else
   return caml_stat_alloc_noexc(sizeof(struct stack_info) +
                                sizeof(value) * wosize +
                                8 /* for alignment */ +
                                sizeof(struct stack_handler));
+#endif
```

Comment:
Could the computation of the size, which is non-trivial and duplicated in this PR, be shared in a single variable before the branches?

Reply:
Do we really want to use `#ifdef __OPENBSD__` here? Naively it would look nicer to me to define a more meaningful variable like USE_MAP_STACK  or something at the toplevel, and test it here (and in the other places that need to be synchronized with it). This would make it easier to extend the scope of systems using MAP_STACK in the future.

Reply:
Both of these comments seem useful to address.

Reply:
Both done.